### ISSUE TITLE
Add early stopping

### ISSUE THREAD

**Issue by ekerazha** [catboost/catboost#43]
At the moment you set an iterations number and you can save the best model (use_best_model parameter). 

However there's no early stopping for the number of iterations. I should be able to stop training after a given number of iterations without a loss/score improvement on the validation set.

Please add early stopping.

**Comment by annaveronika** [comment]
CatBoost has support for early stopping. You can use auto_stop_pval parameter to use overfitting detector.
Here is the description of how it works: https://tech.yandex.com/catboost/doc/dg/concepts/overfitting-detector-docpage/

**Comment by ekerazha** [comment]
@annaveronika 
Thank you for your reply.

I didn't notice that parameter as it's not used in this example: https://tech.yandex.com/catboost/doc/dg/concepts/python-usages-examples-docpage/#custom-loss-function-eval-metric

However, do you think it would be possible to add an early stopping criteria based on the number of iterations (just like in xgboost and lightgbm)?

**Comment by annaveronika** [comment]
Sure, we'll do!

**Comment by annaveronika** [comment]
Issue MATRIXNET-762

**Comment by annaveronika** [comment]
Added a new type "Iter" for overfitting detector. Try updating to the last version.

**Comment by tyokota** [comment]
Will this feature be in the R version soon? I just looked at `catboost.R` and noticed that there were no params set for this option, yet.

**Comment by SRKrish1** [comment]
@ annaveronika, I do not see "Iter" in : https://tech.yandex.com/catboost/doc/dg/concepts/python-reference_catboostregressor-docpage/

**Comment by nikitxskv** [comment]
@tyokota This feature has already been added to R-package.

@SRKrish1 Try to find it on [this](https://tech.yandex.com/catboost/doc/dg/concepts/python-reference_parameters-list-docpage/) page.
